hermes/alertstream_handlers/ingest_from_hop.py

```diff
@@ -4,44 +4,137 @@
 from dateutil.parser import parse, parserinfo
 import logging
 import uuid
+import fastavro
+import healpy as hp
+import numpy as np
+import hashlib
+from astropy.table import Table
+from io import BytesIO
+from django.conf import settings
 
 from hop.io import Metadata
 from hop.models import GCNCircular, JSONBlob
 
-from hermes.models import Message
+from hermes.models import Message, NonLocalizedEvent, NonLocalizedEventSequence
 from hermes import parsers
 
 logger = logging.getLogger(__name__)
 
 GCN_CIRCULAR_PARSER = parsers.GCNCircularParser()
 HERMES_PARSER = parsers.HermesMessageParser()
+IGWN_ALERT_PARSER = parsers.IGWNAlertParser()
 
 TOPIC_PIECES_TO_IGNORE = [
     'gcn.notice',
     'heartbeat'
 ]
 
+
 def should_ingest_topic(topic):
     for topic_piece in TOPIC_PIECES_TO_IGNORE:
         if topic_piece in topic.lower():
             return False
     return True
 
 
-def get_or_create_uuid_for_message(metadata: Metadata) -> uuid.UUID:
+def get_sequence_number(superevent_id: str) -> int:
+    """ This returns the sequence number of the next sequence for a superevent_id. This is a hack to get
+        around the fact that IGWN GWAlerts no longer tell you the sequence number within them.
+    """
+    try:
+        nle = NonLocalizedEvent.objects.get(event_id=superevent_id)
+        return nle.sequences.count() + 1
+    except NonLocalizedEvent.DoesNotExist:
+        return 1  # The nonlocalizedevent doesnt exist in our system yet, so this must be the first sequence
+
+
+def get_skymap_version(superevent_id: str, skymap_hash: uuid) -> int:
+    """ This method gets the most recent previous sequence of this superevent and checks if the skymap has changed.
+        It returns the 'version' of the skymap, which can be used to retrieve the proper file and image files from gracedb.
+    """
+    try:
+        nle = NonLocalizedEvent.objects.get(event_id=superevent_id)
+        latest_sequence = nle.sequences.last()
+        if latest_sequence.skymap_hash != skymap_hash:
+            return latest_sequence.skymap_version + 1
+        return latest_sequence.skymap_version
+    except NonLocalizedEvent.DoesNotExist:
+        return 0  # The nonlocalizedevent doesnt exist in our system yet, so this must be the first skymap version
+
+
+def get_combined_skymap_version(superevent_id: str, skymap_hash: uuid) -> int:
```

Comment:
Is there a way for the hacky-logic in `get_sequence_number()` and (kind of?) duplicated here to exist in only one place? (I'm hoping that in the future, LVK will fix the `igwn.gwalert` format and we'll update the ingester accordingly, ideally, only in that one place where the hacky-logic exists.   If it's not feasible, no big deal, but maybe call out this method as another hack, as was done in `get_sequence_number()`

Reply:
I know they look almost the same but I couldn't think of a good way to break out any of the functionality in a way that reused the code better, since they are a bit different and use different model field references... I will add comments on the hack though.